source/Outfit.cpp

```diff
@@ -88,6 +87,31 @@ namespace {
 				thisMap.erase(it.first);
 		}
 	}
+	
+	// Summation that avoids loss of trailing digits as much as possible.
+	double Sum(const map<double, int> &partialTerms)
```

Comment:
The memory usage for attributes increases quite a bit using this solution; we now get an entry in the partialTerms map for every fraction that was added to the attribute. This will happen for every ship, including ships where the outfits are typically not changed.
If we go this route, then there might be another solution that uses an equal amount of memory, but that computes faster and that is to use very high precision numbers instead of the partial terms.

Another consideration is that we only need this very high precision when we are actually changing ship outfits, and the raw data to generate the high precision already seems to be stored in the outfit-counts on a ship. We should be able to generate the high precision data on-the-fly when we need them during Outfitting (Thus reducing the overal memory usage at the costs of additional CPU time when we are starting to change Outfits.)